language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/sourceprune/CompletionsTokenTraverserFactory.java

```diff
@@ -0,0 +1,117 @@
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.compiler.workspace.WorkspaceDocumentException;
+import org.ballerinalang.langserver.compiler.workspace.WorkspaceDocumentManager;
+import org.ballerinalang.langserver.sourceprune.SourcePruneContext;
+import org.ballerinalang.langserver.sourceprune.TokenTraverser;
+import org.ballerinalang.langserver.sourceprune.TokenTraverserFactory;
+import org.wso2.ballerinalang.compiler.parser.antlr4.BallerinaParser;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Factory class for providing token traversers and related artifacts for 'completions' service operation.
+ */
+public class CompletionsTokenTraverserFactory implements TokenTraverserFactory {
+    private final SourcePruneContext sourcePruneCtx;
+    private final boolean pruneTokens;
+    private final TokenStream tokenStream;
+
+    private static final List<Integer> BLOCK_REMOVE_KW_TERMINALS;
+    private static final List<Integer> LHS_TRAVERSE_TERMINALS;
+    private static final List<Integer> RHS_TRAVERSE_TERMINALS;
+
+    static {
+        BLOCK_REMOVE_KW_TERMINALS = Arrays.asList(
+                BallerinaParser.SERVICE, BallerinaParser.FUNCTION, BallerinaParser.TYPE, BallerinaParser.MATCH,
+                BallerinaParser.FOREACH, BallerinaParser.WORKER
+        );
+        LHS_TRAVERSE_TERMINALS = Arrays.asList(
+                BallerinaParser.LEFT_BRACE, BallerinaParser.RIGHT_BRACE, BallerinaParser.SEMICOLON,
+                BallerinaParser.COMMA, BallerinaParser.LEFT_PARENTHESIS, BallerinaParser.RIGHT_PARENTHESIS,
+                BallerinaParser.LT, BallerinaParser.RETURNS, BallerinaParser.TRANSACTION,
+                BallerinaParser.LEFT_CLOSED_RECORD_DELIMITER, BallerinaParser.LEFT_BRACKET
+        );
+        RHS_TRAVERSE_TERMINALS = Arrays.asList(
+                BallerinaParser.SEMICOLON, BallerinaParser.DocumentationLineStart,
+                BallerinaParser.AT, BallerinaParser.LEFT_BRACE, BallerinaParser.RIGHT_BRACE,
+                BallerinaParser.RIGHT_PARENTHESIS, BallerinaParser.IMPORT, BallerinaParser.GT,
+                BallerinaParser.XMLNS, BallerinaParser.SERVICE, BallerinaParser.PUBLIC, BallerinaParser.PRIVATE,
+                BallerinaParser.REMOTE, BallerinaParser.FUNCTION, BallerinaParser.TYPE, BallerinaParser.ANNOTATION,
+                BallerinaParser.CONST, BallerinaParser.RIGHT_BRACKET, BallerinaParser.RIGHT_CLOSED_RECORD_DELIMITER,
+                BallerinaParser.RESOURCE, BallerinaParser.LISTENER, BallerinaParser.MATCH, BallerinaParser.IF,
+                BallerinaParser.WHILE, BallerinaParser.FOREACH, BallerinaParser.BREAK, BallerinaParser.BREAK,
+                BallerinaParser.FORK, BallerinaParser.THROW, BallerinaParser.TRANSACTION, BallerinaParser.WORKER
+        );
+    }
+
+    public CompletionsTokenTraverserFactory(Path filePath, WorkspaceDocumentManager documentManager,
```

Comment:
Shall we add a doc comment for the public method?

Reply:
Added missing doc comments in other places. This is a constructor which we usually don't add doc comments since class level doc comment is already there. For the sake of consistency let's keep this as it is. WDYT?